src/main/plugins/org.dita.base/build_preprocess_template.xml

```diff
@@ -54,7 +54,15 @@ See the accompanying LICENSE file for applicable license.
     
     <basename property="dita.input.filename" file="${args.input}" />
     <basename property="dita.map.filename.root" file="${dita.input.filename}" suffix=".ditamap" />
-    <basename property="dita.topic.filename.root" file="${dita.input.filename}" suffix=".dita" />
+    <condition property="dita.topic.filename.extension" value=".xml">
```

Comment:
I don't like hard-coding common extensions for topics and then failing to remove the extension if some other extension is used. This should be handled with more generic approach, one which supports any extension.